demos/serverless/src/handlers.js

```diff
@@ -272,3 +274,27 @@ function response(statusCode, contentType, body) {
     isBase64Encoded: false
   };
 }
+
+function addSignalMetricsToCloudWatch(logMsg, meetingId, attendeeId) {
+  const logMsgJson = JSON.parse(logMsg);
+  const metricList = ['signalingOpenDurationMs'];
+  for (let metricIndex = 0; metricIndex <= metricList.length; metricIndex += 1) {
+    const metricName = metricList[metricIndex];
+    if (logMsgJson.attributes.hasOwnProperty(metricName)) {
+      const metricValue = logMsgJson.attributes[metricName];
+      console.log('Logging metric -> ', metricName, ': ', metricValue );
+      const putMetric =
+        metricScope(metrics => async (metricName, metricValue, meetingId, attendeeId) => {
+          metrics.putDimensions({MId: meetingId, AId: attendeeId});
+          metrics.putMetric(metricName, metricValue);
+        });
+      putMetric(metricName, metricValue, meetingId, attendeeId);
+
+      const putSignalMetric =
```

Comment:
- Create `putSignalMetric` outside the `for` loop. Correct me if this library requires us to instantiate per single call.
- `async` is unnecessary.

Reply:
Yes, one call per metricName and metricValue

Reply:
The following five parameters are available in the function scope you are passing to `metricScope`. i.e., None of these parameters comes from the outside of the function scope.
- `metrics`, `metricName`, `metricValue`, `meetingId`, and `attendeeId`

```
const putMetric =
  metricScope(metrics => (metricName, metricValue, meetingId, attendeeId) => {
    metrics.putDimensions({MeetingId: meetingId, AttendeeId: attendeeId});
    metrics.putMetric(metricName, metricValue);
  });
```

Can you declare this function only once outside the "for" loop like the following?

```
const putMetric =
  metricScope(metrics => (metricName, metricValue, meetingId, attendeeId) => {
    metrics.putDimensions({MeetingId: meetingId, AttendeeId: attendeeId});
    metrics.putMetric(metricName, metricValue);
  });

for (let metricIndex = 0; metricIndex <= metricList.length; metricIndex += 1) {
    const metricName = metricList[metricIndex];
    if (logMsgJson.attributes.hasOwnProperty(metricName)) {
      const metricValue = logMsgJson.attributes[metricName];
      console.log('Logging metric -> ', metricName, ': ', metricValue );
      
      // Use a generated function here
      putMetric(metricName, metricValue, meetingId, attendeeId);
      ...
```

Reply:
Brought the putMetric initialization outside the for loop